### PR TITLE
np-47689: Use field verified to know if we should keep id or not on p…

### PR DESCRIPTION
**Pull request by torarnet**
…roject contributor

we only want id on project contributors that are verified in cristin to make it easier for frontend

### CODE REVIEW

File: cristin-project/src/main/java/no/unit/nva/cristin/projects/model/cristin/adapter/CristinPersonToPerson.java [hunk 1]
```diff
@@ -19,15 +20,20 @@ public Person apply(CristinPerson cristinPerson) {
...
             cristinPerson.firstName(),
             cristinPerson.surname(),
             cristinPerson.email(),
             cristinPerson.phone());
     }
 
-    private URI convertId(String cristinPersonId) {
-        return getNvaApiId(cristinPersonId, PERSON_PATH_NVA);
+    private URI convertId(Boolean identified, String cristinPersonId) {
+        return identifiedFieldNotPresent(identified) || identified
```
Comment (brinxmat) [inline cristin-project/src/main/java/no/unit/nva/cristin/projects/model/cristin/adapter/CristinPersonToPerson.java:31]: So, if we don't have the data, or the person is identifier…we keep the URI?

**Review by brinxmat — DISMISSED** [review]

File: cristin-project/src/main/java/no/unit/nva/cristin/projects/model/cristin/adapter/CristinPersonToPerson.java (hunk 1 above)
Comment (torarnet) [reply cristin-project/src/main/java/no/unit/nva/cristin/projects/model/cristin/adapter/CristinPersonToPerson.java:31]: Exactly. We dont want to remove the id if the field is missing. This makes it backwards compatible and if we want to enforce the field later it is a simple change in proxy

**Review by StigNorland — APPROVED** [review]
this is a no comments, comment

**Review by torbjokv — DISMISSED** [review]

File: cristin-project/src/main/java/no/unit/nva/cristin/projects/model/cristin/adapter/CristinPersonToPerson.java [hunk 2]
```diff
@@ -19,15 +20,20 @@ public Person apply(CristinPerson cristinPerson) {
...
+            convertId(cristinPerson.identifiedCristinPerson(), cristinPerson.cristinPersonId()),
             cristinPerson.firstName(),
             cristinPerson.surname(),
             cristinPerson.email(),
             cristinPerson.phone());
     }
 
-    private URI convertId(String cristinPersonId) {
-        return getNvaApiId(cristinPersonId, PERSON_PATH_NVA);
+    private URI convertId(Boolean identified, String cristinPersonId) {
```
Comment (LarsV123) [inline cristin-project/src/main/java/no/unit/nva/cristin/projects/model/cristin/adapter/CristinPersonToPerson.java:30]: Nitpick / non-blocking:
I needed some time and your comment below to understand what this does and why.
Can you think if a better structure and/or method name here, to make it more obvious?

The only suggestion I can think of is this, but I'm not sure it's actually better:
```
private URI getPersonUri(Boolean identified, String cristinPersonId) {
    if (isNull(identified) || identified) {
        return getNvaApiId(cristinPersonId, PERSON_PATH_NVA);
    }
    return null;
}
```

Comment (torarnet) [reply cristin-project/src/main/java/no/unit/nva/cristin/projects/model/cristin/adapter/CristinPersonToPerson.java]: Updated PR

**Review by brinxmat — APPROVED** [review]

**Pull request merged**